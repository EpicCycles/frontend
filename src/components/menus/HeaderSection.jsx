import React from "react";
import {Link} from "react-router-dom";

const HeaderSection = props => (
    props.sectionContents.map(group => {
        return (
            <li className="dropdown" key={'navSection' + group.groupPos}>
                <p className="dropbtn">{group.groupHeader}</p>
                <div className="dropdown-content">
                    <div className="dropdown-section">{group.groupHeader}</div>
                    {group.groupLinks.map(linkData => {
                        return <Link
                            key={`link${linkData.linkNumber}`}
                            className="dropbtn"
                            to={linkData.linkRoute ? linkData.linkRoute : linkData.linkURL}
                        >
                            {linkData.displayText}
                        </Link>
                    })}
                </div>
            </li>
        )
    })
);
export default HeaderSection;

