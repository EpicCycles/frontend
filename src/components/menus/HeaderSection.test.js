import HeaderSection from './HeaderSection';

test('it displays correctly', () => {
  const sectionContents = [
    {
      groupHeader: 'Customer',
      groupLinks: [
        { displayText: 'Find Customer', linkRoute: '/customer-search', linkNumber: 1 },
        { displayText: 'New Customer', linkRoute: '/customer', linkNumber: 2 },
      ],
    },
    {
      groupHeader: 'Core Data',
      groupLinks: [
        { displayText: 'Quote Sections', linkRoute: '/framework', linkNumber: 3 },
        { displayText: 'Brands', linkRoute: '/brands', linkNumber: 4 },
      ],
    },
  ];
  const component = shallow(<HeaderSection sectionContents={sectionContents} />);
  expect(toJson(component)).toMatchSnapshot();
});
