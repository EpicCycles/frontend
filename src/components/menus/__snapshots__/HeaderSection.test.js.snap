// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it displays correctly 1`] = `
Array [
  <li
    className="dropdown"
    key="navSectionundefined"
  >
    <p
      className="dropbtn"
    >
      Customer
    </p>
    <div
      className="dropdown-content"
    >
      <div
        className="dropdown-section"
      >
        Customer
      </div>
      <ForwardRef
        className="dropbtn"
        key="link1"
        to="/customer-search"
      >
        Find Customer
      </ForwardRef>
      <ForwardRef
        className="dropbtn"
        key="link2"
        to="/customer"
      >
        New Customer
      </ForwardRef>
    </div>
  </li>,
  <li
    className="dropdown"
    key="navSectionundefined"
  >
    <p
      className="dropbtn"
    >
      Core Data
    </p>
    <div
      className="dropdown-content"
    >
      <div
        className="dropdown-section"
      >
        Core Data
      </div>
      <ForwardRef
        className="dropbtn"
        key="link3"
        to="/framework"
      >
        Quote Sections
      </ForwardRef>
      <ForwardRef
        className="dropbtn"
        key="link4"
        to="/brands"
      >
        Brands
      </ForwardRef>
    </div>
  </li>,
]
`;
